### ISSUE TITLE
Support Menu Contributions model element

### ISSUE THREAD

**Issue by vogella** [eclipse-efx/efxclipse-rt#274]
Currently, the FX render seem to not support "Menu Contributions" similar to the SWT renderers.

If you define a Menu Contribution extending the correct Parent-ID the SWT Renderer will add this menu to the menu of the Window. The FX renderer seems to be missing that capability. 

To test, simply generate a new e4 FX application using the template. Afterwards change the ID of the menu to "org.eclipse.ui.main.menu". Create a new plug-in via the "Menu contribution using 4.x API" PDE template. Remove the jface reference from the manifest and change the generated handler to the following.

```
@Execute
	public void execute() {System.out.println("Hello");}
````

**Comment by tomsontom** [comment]
Yes and (Menu|Toolbar|Trim)Contributions are a concept introduce inappropriately to the e4 model to support the compat layer we don't support them because we by definition don't have a compat layer to support.

What usecase do you have for them you can not solve by directly contributing the elements?

**Comment by vogella** [comment]
> What usecase do you have for them you can not solve by directly contributing the elements?

The use case im aware of is dynamically contributing to any window. I don't think without Menu Contributions or model processors this is possible. Just tested in FX and if I have two windows with a main menu "org.eclipse.ui.main.menu" and contribute a normal menu to it, only the first window gets the additional menu entry.

**Comment by tomsontom** [comment]
I thought the new xpath contribution system allows this kind of thing as well, but that would naturally only work for statically defined Windows. Does the SWT-Version also contribute those items if you construct windows at runtime?

**Comment by vogella** [comment]
>Does the SWT-Version also contribute those items if you construct windows at runtime?

Yes.
